### PR TITLE
api authentication

### PR DESCRIPTION
**Pull request by mohanqxf2**
Created a new api endpoint for add articles with API key authentication.
- 
- on the routes.py file, in add_articles function I have removed line "_if addarticlesform.submit.data:_" which was causing issues to  add articles to database through API call. 
- removing the _if condition_ does not make the data validation through web page

```
 if request.method == 'POST':
            -if addarticlesform.submit.data:-
                  article = Articles(addarticlesform.url.data,addarticlesform.....)
                  db.session.add(article)
```
            
-

### CODE REVIEW

File: newsletter_automation/conf/apikey_conf.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""
```
Comment (qxf2) [inline newsletter_automation/conf/apikey_conf.py:1]: This file should not be part of version control. Add it to .gitignore and then put instructions in the README as setup instructions.

File: newsletter_automation/helpers/authentication_required.py [hunk 2]
```diff
@@ -3,7 +3,11 @@
 """
 
 from functools import wraps
-from flask import render_template, session
+from flask import render_template, session, request, jsonify
+import conf.apikey_conf as conf
```
Comment (qxf2) [inline newsletter_automation/helpers/authentication_required.py:7]: The variable name `conf` is too general. Further, `conf` is also the name of a directory. Import it at as something else. Your code should be readable. Right now, lines like `API = conf.API_KEY` does not make sense especially if we know `conf` is a directory name.

File: newsletter_automation/newsletter/routes.py [hunk 3]
```diff
@@ -92,17 +92,14 @@ def index():
...
-@app.route('/articles', methods=['GET', 'POST'])
-@Authentication_Required.requires_auth
-def articles():
+def add_articles():
     "This page adds articles to the database"
     try:
         addarticlesform = AddArticlesForm(request.form)
         category = AddArticlesForm(request.form)
         if request.method == 'POST':
-            if addarticlesform.submit.data:
```
Comment (qxf2) [inline newsletter_automation/newsletter/routes.py:103]: What is the impact of removing `if addarticlesform.submit.data`? You explained why you need to do it. But what is the impact on the existing web interface?

**Review by qxf2 — CHANGES_REQUESTED** [review]
@mohanqxf2 - I have added some basic comments. Please make those changes. When you raise a PR, please add notes on how you tested your change. You should address what you tested for your own work and how you ensured there is no regression on the /articles route of the web interface.

File: newsletter_automation/newsletter/routes.py (hunk 3 above)
Comment (mohanqxf2) [reply newsletter_automation/newsletter/routes.py:103]: The impact of removing `if addarticlesform.submit.data` on the existing web interface would be, URL's would not be validated(if given URL is valid URL or not) from wtforms _form.py_ ` url = URLField('url', validators=[DataRequired(), url()])
`

**Comment by mohanqxf2** [comment]
Here are my observations on removing `if addarticlesform.submit.data` condition
https://docs.google.com/spreadsheets/d/1GquosS2pAEJbSZfOn279y4Yhe88RenRLnNVz_YWPedA/edit?usp=sharing

**Pull request merged**